plugin/src/test/java/io/jenkins/plugins/analysis/warnings/PluginArchitectureTest.java

```diff
@@ -60,4 +67,32 @@
                             .or(have(simpleNameStartingWith("Integration")))
                             .or(have(simpleName("ToolsLister")))))
                     .should().dependOnClassesThat().resideInAnyPackage("org.junit");
+
+    @ArchTest
+    static final ArchRule JAVASCRIPTMETHOD_PUBLIC_CLASS =
+            methods().that().areAnnotatedWith(JavaScriptMethod.class)
+                    .should().beDeclaredInClassesThat().arePublic();
+
+    @ArchTest
+    static final ArchRule EXTENSION_PUBLIC_CLASS =
+            classes().that().areAnnotatedWith(Extension.class)
+                    .should().bePublic();
+
+    @ArchTest
+    static final ArchRule DATABOUND_PUBLIC_CLASS_AND_CONSTRUCTOR =
+            constructors().that().areAnnotatedWith(DataBoundConstructor.class)
+                    .should().bePublic()
+                    .andShould().beDeclaredInClassesThat().arePublic();
+
+    @ArchTest
+    static final ArchRule JAVASCRIPTMETHOD_AND_DATABOUND_PUBLIC_METHOD =
+            methods().that().areAnnotatedWith(JavaScriptMethod.class)
+                    .or().areAnnotatedWith(DataBoundSetter.class)
+                    .should().bePublic();
+
+    @ArchTest
+    static final ArchRule READRESOLVE_SERIALIZABLE_CLASS =
+            methods().that().haveFullName("readResolve").and().haveRawReturnType(Object.class)
```

Comment:
Sonst geht es bei mir nicht:

```suggestion
            methods().that().haveName("readResolve").and().haveRawReturnType(Object.class)
```

Haben Sie die Tests auch mal ansprechen lassen? Das geht bei mir aktuell im Warnings Plugin gar nicht, die schlagen nie an. Ist das ein ArchUnit Bug?

Reply:
Hallo @uhafner, komisch, wir schauen schnellstmöglich nach und melden uns dann bei Ihnen!

Reply:
Im CI funktionieren die Tests, auch in meiner virtuellen Maschine. Seltsam, was da auf meinem Rechner nicht passt.

Reply:
Hallo @uhafner , also ich habe gerade auch nochmal alle Tests durchlaufen lassen und bei mir läuft auch alles durch (Windows))

Reply:
Das ist ja das Problem, Sie sollen ja fehlschlagen, wenn man sich nicht an eine Regel hält. Ich habe eine Klasse verändert, so dass die Regel anschlagen sollte. Sie tut es aber nicht... 

Ich habe mal einen Bug aufgemacht unter https://github.com/TNG/ArchUnit/issues/395